ai-catalog/samples/gemini-live-todo/src/main/java/com/android/ai/samples/geminilivetodo/ui/TodoScreen.kt

```diff
@@ -86,11 +91,21 @@ import kotlin.collections.reversed
 fun TodoScreen(viewModel: TodoScreenViewModel = hiltViewModel()) {
     val uiState by viewModel.uiState.collectAsStateWithLifecycle()
     var text by remember { mutableStateOf("") }
+    val context = LocalContext.current
 
-    val activity = LocalActivity.current as Activity
+    val requestPermissionLauncher = rememberLauncherForActivityResult(
+        contract = ActivityResultContracts.RequestPermission(),
+        onResult = { isGranted: Boolean ->
+            if (isGranted) {
+                viewModel.toggleLiveSession()
+            } else {
+                Toast.makeText(context, R.string.error_permission, Toast.LENGTH_SHORT).show()
```

Comment:
optional, but consider using Snackbar here instead of Toast. I think material design guidelines recommend snackbars instead